tests/e2e/no_hci_mesh_rwx_singlevc_topology_disruptive.go

```diff
@@ -0,0 +1,726 @@
+						_, err = vm.PowerOn(ctx)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+						err = vm.WaitForPowerState(ctx, vimtypes.VirtualMachinePowerStatePoweredOn)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					}
+				}
+			}
+		}()
+
+		ginkgo.By("Scale up deployment set replica count of dep1, dep2 and dep3 to 10 when Az2 site is down")
+		replica = 10
+		for i := 0; i < 3; i++ {
+			_, _, _ = createVerifyAndScaleDeploymentPods(ctx, client, namespace, replica,
+				true, labelsMap, pvclaim, nodeSelectorTerms, execRWXCommandPod, nginxImage, true, depl[i], 0)
+		}
+
+		ginkgo.By("Scale down deployment set replica count of dep4, dep5 and dep6 to 4 when Az2 site is down")
+		replica = 4
+		for i := 3; i < 6; i++ {
+			_, _, _ = createVerifyAndScaleDeploymentPods(ctx, client, namespace, replica,
+				true, labelsMap, pvclaim, nodeSelectorTerms, execRWXCommandPod, nginxImage, true, depl[i], 0)
+		}
+
+		ginkgo.By("Create StatefulSet with replica count 3 with RWX PVC access mode")
+		service, _, _ = createStafeulSetAndVerifyPVAndPodNodeAffinty(ctx, client, namespace, true,
+			int32(stsReplicas), false, nil, false, false, false, accessmode, storageclass, false, "")
+
+		ginkgo.By("Create new rwx pvc when Az2 site is down")
+		createPvcItr = 5
+		_, pvclaimsNew, err := createStorageClassWithMultiplePVCs(client, namespace, labelsMap, scParameters,
+			diskSize, nil, bindingModeImm, false, accessmode, "", storageclass, createPvcItr, true, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			for i := 0; i < len(pvclaimsNew); i++ {
+				pv := getPvFromClaim(client, pvclaimsNew[i].Namespace, pvclaimsNew[i].Name)
+				err = fpv.DeletePersistentVolumeClaim(ctx, client, pvclaimsNew[i].Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Create standalone Pods with different read/write permissions and attach it to a newly created rwx pvcs")
+		noPodsToDeploy = 2
+		for i := 0; i < len(pvclaimsNew); i++ {
+			pods, err := createStandalonePodsForRWXVolume(client, ctx, namespace, nil, pvclaimsNew[i], false, execRWXCommandPod,
+				noPodsToDeploy)
+			podList = append(podList, pods...)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		}
+		defer func() {
+			for i := 0; i < len(podList); i++ {
+				ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", podList[i].Name, namespace))
+				err = fpod.DeletePodWithWait(ctx, client, podList[i])
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		// Bring up
+		ginkgo.By("Bring up all ESXi host which were powered off")
+		for i := 0; i < len(powerOffHostsList); i++ {
+			powerOnEsxiHostByCluster(powerOffHostsList[i])
+		}
+
+		ginkgo.By("Bring up all K8s nodes which got disconnected due to powered off esxi hosts")
```

Comment:
write a util to use wherever required to check k8s nodes

Reply:
Ack